libnavigation-ui/src/main/java/com/mapbox/navigation/ui/camera/NavigationCamera.java

```diff
@@ -0,0 +1,580 @@
+      listener.onTransitionFinished(trackingCameraMode);
+    }
+  }
+
+  void updateTransitionListenersCancelled(@CameraMode.Mode int cameraMode) {
+    Integer trackingCameraMode = findTrackingModeFor(cameraMode);
+    if (trackingCameraMode == null) {
+      return;
+    }
+    for (OnTrackingModeTransitionListener listener : onTrackingModeTransitionListeners) {
+      listener.onTransitionCancelled(trackingCameraMode);
+    }
+  }
+
+  @Nullable
+  Integer findTrackingModeFor(@CameraMode.Mode int cameraMode) {
+    if (cameraMode == CameraMode.TRACKING_GPS) {
+      return NAVIGATION_TRACKING_MODE_GPS;
+    } else if (cameraMode == CameraMode.TRACKING_GPS_NORTH) {
+      return NAVIGATION_TRACKING_MODE_NORTH;
+    } else if (cameraMode == CameraMode.NONE) {
+      return NAVIGATION_TRACKING_MODE_NONE;
+    } else {
+      return null;
+    }
+  }
+
+  void updateIsResetting(boolean isResetting) {
+    this.isCameraResetting = isResetting;
+  }
+
+  private void initializeWith(MapboxNavigation navigation) {
+//    navigation.setCameraEngine(new DynamicCamera(mapboxMap));
+    updateCameraTrackingMode(trackingCameraMode);
+  }
+
+  private void tryToBuildRouteInformationAndAdjustCamera() {
+    if (isTrackingEnabled()) {
+      currentRouteInformation = new RouteInformation(currentRouteProgress.route(), currentLocation, currentRouteProgress);
+      if (!isCameraResetting) {
+        adjustCameraFromLocation(currentRouteInformation);
+      }
+    }
+  }
+
+  /**
+   * Creates a camera position based on the given route.
+   * <p>
+   * From the {@link DirectionsRoute}, an initial bearing and target position are created.
+   * Then using a preset tilt and zoom (based on screen orientation), a {@link CameraPosition} is built.
+   *
+   * @param route used to build the camera position
+   * @return camera position to be animated to
+   */
+  @NonNull
+  private RouteInformation buildRouteInformationFromRoute(DirectionsRoute route) {
+    return new RouteInformation(route, null, null);
+  }
+
+  /**
+   * Creates a camera position based on the given location.
+   * <p>
+   * From the {@link Location}, a target position is created.
+   * Then using a preset tilt and zoom (based on screen orientation), a {@link CameraPosition} is built.
+   *
+   * @param location used to build the camera position
+   * @return camera position to be animated to
+   */
+  @NonNull
+  private RouteInformation buildRouteInformationFromLocation(@Nullable Location location, @Nullable RouteProgress routeProgress) {
+    return new RouteInformation(null, location, routeProgress);
+  }
+
+  private void onCameraTransitionFinished() {
+    if (isCameraResetting && currentRouteInformation != null) {
+      adjustCameraForReset(currentRouteInformation);
+    }
+  }
+
+  private void animateCameraForRouteOverview(RouteInformation routeInformation, int[] padding) {
+//    Camera cameraEngine = navigation.getCameraEngine();
```

Comment:
Same as above, why we commented out this?

Same question for the other commented out parts in `NavigationCamera` 👇

Reply:
`Camera` doesn't live in `MapboxNavigation` anymore.

Reply:
Same as above. Now that is going to live under `libnavigation-ui` we should revisit `Camera` implementation. Let's cut a ticket to track this work.